openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/LightGBMBinaryClassificationModelTest.java

```diff
@@ -236,4 +249,122 @@ private boolean compareDoubles(final double a, final double b, final int exclude
         );
     }
 
+    /**
+     * Compares files with the same name across two folders.
+     * If there are any differences in the models within such folders they are reported
+     * in a colored report. Requires python3 with termcolor installed for best readability.
+     *
+     * @param referenceModelsFolder Folder with reference models
+     * @param newModelsFolder       Folder with new models to compare to reference
+     * @return true if there are no differences and all the dependencies are installed, false otherwise.
+     * @throws IOException          In case the process errors.
+     * @throws InterruptedException In case awaiting for the process to finish fails.
+     * @since 1.0.19
+     */
+    private boolean compareModelFilesAndDoPrettyReport(
+            final String referenceModelsFolder,
+            final String newModelsFolder) throws IOException, InterruptedException {
+
+        // Check for differences and report in detailed manner!
+        final Process p = new ProcessBuilder(
+                "python",
+                "src/test/resources/diff_models.py",
+                referenceModelsFolder,
+                newModelsFolder
+        ).start();
+
+        // Fetch process output
+        final BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
```

Comment:
We should use `try-with-resources` here in order to make sure that the file is securely closed

see https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
If you need help to change this let me know